hydra-node/test/Hydra/Model.hs

```diff
@@ -222,16 +222,16 @@ instance StateModel WorldState where
     (from tx, value tx) `List.elem` confirmedUTxO offChainState
   precondition _ Wait{} =
     True
-  precondition WorldState{hydraState = Open{}} ObserveConfirmedTx{} =
-    True
+  precondition WorldState{hydraState = Open{offChainState = OffChainState{seenTransactions}}} (ObserveConfirmedTx tx) =
+    isJust $ Map.lookup tx seenTransactions
```

Comment:
Could: add a note why this is needed.

I do understand that this is important in shrinking when we only can keep this action if the corresponding `NewTx` result can be found in the `seenTransactions`.

Reply:
Actually, this should disappear as it won't be needed anymore thanks to the use of `Var`iables